codecarbon/emissions_tracker.py

```diff
@@ -698,19 +702,18 @@ def _measure_power_and_energy(self) -> None:
         self._last_measured_time = time.time()
         self._measure_occurrence += 1
         if (
-            self._cc_api__out is not None or self._cc_prometheus_out is not None
-        ) and self._api_call_interval != -1:
-            if self._measure_occurrence >= self._api_call_interval:
-                emissions = self._prepare_emissions_data(delta=True)
-                logger.info(
-                    f"{emissions.emissions_rate * 1000:.6f} g.CO2eq/s mean an estimation of "
-                    + f"{emissions.emissions_rate*3600*24*365:,} kg.CO2eq/year"
-                )
-                if self._cc_api__out:
-                    self._cc_api__out.out(emissions)
-                if self._cc_prometheus_out:
-                    self._cc_prometheus_out.out(emissions)
-                self._measure_occurrence = 0
+            len(self._output_handlers) > 0
+            and self._api_call_interval != -1
+            and self._measure_occurrence >= self._api_call_interval
+        ):
+            emissions = self._prepare_emissions_data(delta=True)
+            logger.info(
+                f"{emissions.emissions_rate * 1000:.6f} g.CO2eq/s mean an estimation of "
+                + f"{emissions.emissions_rate*3600*24*365:,} kg.CO2eq/year"
+            )
+            for handler in self._output_handlers:
+                handler.out(emissions)
```

Comment:
Hey @eledhwen  thanks! Sorry for my late reply on this. I completely missed this PR.
I like this idea. The problem I see is if here you do `handler.out(emissions)` you will output the emissions in delta mode in all output_handlers.  So as you suggested in [here]( https://github.com/mlco2/codecarbon/pull/458#issuecomment-1752002391), we need to flag the output_methods as `live` and `flush`/`exit`. Or some other kind of flagging.

In  this [PR](https://github.com/mlco2/codecarbon/pull/542/files#) I did something similar by adding `use_emissions_delta`  true/false to each output method (By default as `false` in the BaseOutput`). 

On the other hand `_output_handlers` only contains for now prometheus and the api, so we could merge this and do the flag in a separate PR. What do you think?

Reply:
Indeed, I like the idea of the methods better. And if not implemented, the output method will do nothing.
> In any case, it's a breaking change on the part of the BaseOutput API ... 

Not necessarily! If we keep having and `out` method that points by default to the `exit` method for instance (as a first version). Then we could deprecate this `out` method in favor of the others on the long run. 

In any case, I think this PR can be merged

Reply:
I'll try to rebase and have a shot at it asap

Reply:
I just pushed a new draft, I didn't look into tests nor documentation, let me know what you think.

Some notes:
* `BaseOutput` has 3 separate methods, `out`, `live_out` and `task_out`
* removal of `persistence_objs` which is now the same set of objects as `_output_handlers`
* I took some liberties with the `_prepare_emissions_data` and separated the aggregation from the delta computation, steering them toward a more SRP-compatible design
* I took some liberties with the time at which some emission data are computed, I preferred more straightforward code to slight optimizations, but this can be challenged